x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/external/action/openai/OpenAiActionCreator.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.inference.external.action.openai;
+
+import org.elasticsearch.xpack.inference.external.action.ExecutableAction;
+import org.elasticsearch.xpack.inference.external.http.sender.Sender;
+import org.elasticsearch.xpack.inference.logging.ThrottlerManager;
+import org.elasticsearch.xpack.inference.services.openai.embeddings.OpenAiEmbeddingsModel;
+
+import java.util.Map;
+import java.util.Objects;
+
+public class OpenAiActionCreator implements OpenAiActionVisitor {
```

Comment:
I'm using the visitor pattern here to avoid needing to use a switch-case in a few places within the `OpenAiService`. Essentially this will rely on the model type to determine what type of action to be created and execute for sending a request (e.g. text embedding model, completion model etc).

This class will have to include a `create` method for each type of open ai model created in the future. The task settings will be different depending on whether we're doing text embedding or completion or other things.